tools/c7n_index/c7n_index/metrics.py

```diff
@@ -33,6 +50,10 @@
     'type': 'object',
     'additionalProperties': True,
     'properties': {
+        'indexer': {
+            'type': 'object',
```

Comment:
any chance you could take a stab at putting the actual fields in use, the schema helps serve as documentation, although perhaps also dropping a sanitized config example.

for json schema this might help (oneOf for the different types)
https://spacetelescope.github.io/understanding-json-schema/reference/combining.html